### PR TITLE
Micro-optimization of `LazyIndexImpl.get()`

### PR DESCRIPTION
**Pull request by snazy**
Eliminates an `instanceof` in the usually executed code path.

### CODE REVIEW

**Review by adutra — DISMISSED** [review]

File: versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java [hunk 1]
```diff
@@ -41,27 +42,28 @@ private NonLockingSupplyOnce(Supplier<T> loader) {
...
-    private T eval(Object i) {
-      if (i instanceof RuntimeException) {
-        throw (RuntimeException) i;
+    private T eval() {
+      Object r = result;
+      if (r != null) {
+        return (T) r;
       }
-      return (T) i;
+      throw ex;
```
Comment (dimas-b) [inline versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java:66]: I'm not sure the order of assignments between `results`, `ex` and `loaded` is actually guaranteed to be the the same in another thread :thinking:

Comment (snazy) [reply versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java:66]: This thing is not thread safe and not used in such a case

Comment (adutra) [reply versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java:66]: There wasn't any synchronization in this class, so I am assuming that it is not meant to be thread-safe. Even the previous version would have had issues if `get()` is accessed concurrently.

**Comment by snazy** [comment]
Ugh - forgot that it's meant to lazy-load even `null` values.

File: versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java [hunk 2]
```diff
@@ -30,38 +30,40 @@ public static <T> Supplier<T> memoize(Supplier<T> loader) {
   }
 
   private static final class NonLockingSupplyOnce<T> implements Supplier<T> {
-    private final Supplier<T> loader;
+    private int loaded;
     private Object result;
```
[the remark this answers is not in the document]
Comment (snazy) [reply versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java:34]: Let's keep the object small. Exceptions are an exception ;)

**Review by adutra — APPROVED** [review]

**Pull request merged**